### PR TITLE
QMAPS-534 [ui] Add ScaleControl

### PR DESCRIPTION
**Pull request by patfrat**
Adding Scale Control (meter unit) in the copyright panel for desktop environment
Scale control is placed on the bottom right under controls in responsive environment

### CODE REVIEW

**Comment by patfrat** [comment]
min-width is now before max-width for the media queries order
scss simplified too but i keep the nested declaration and the & operand

File: src/mapbox/extended_nav_control.js [hunk 1]
```diff
@@ -31,6 +34,20 @@ export default class ExtendedControl {
     this._map.on('rotate', _pitchAndRotateCompassArrow)
     this._map.on('pitch', _pitchAndRotateCompassArrow)
 
+    this.scaleAttibutionContainer =  document.createElement('div')
```
Comment (patfrat) [inline src/mapbox/extended_nav_control.js]: rename it to scaleAtt(r)ibution

File: src/mapbox/extended_nav_control.js [hunk 2]
```diff
@@ -31,6 +34,20 @@ export default class ExtendedControl {
...
+    this.scaleAttibutionContainer.className = 'map_control__scale_attribute_container'
```
Comment (patfrat) [inline src/mapbox/extended_nav_control.js]: rename it to scaleAtt(r)ibution

Comment (patfrat) [reply src/mapbox/extended_nav_control.js]: and so on...

File: package.json [hunk 3]
```diff
@@ -53,7 +53,7 @@
     "json-loader": "0.5.7",
     "klokantech-noto-sans": "git+https://github.com/QwantResearch/fonts.git#f25a802",
     "mapbox-gl": "0.49.0",
-    "mapbox-gl-js-mock": "https://github.com/QwantResearch/mapbox-gl-js-mock.git#58d80b5",
+    "mapbox-gl-js-mock": "https://github.com/QwantResearch/mapbox-gl-js-mock.git#QMAPS-534_scale_control",
```
Comment (patfrat) [inline package.json]: only if the corresponding branch is kept

Comment (Julien-laville) [reply package.json]: Yep, the process is when all is validated: merge dependencies, update git tag, test again, merge this.

Comment (G-Ray) [inline package.json]: don't forget to merge the PRs on 'mapbox-gl-js-mock.git' and fix the right commit on master once it's merged

File: src/adapters/scene.js [hunk 4]
```diff
@@ -30,6 +30,7 @@ function Scene() {
 
 Scene.prototype.initMapBox = function () {
   this.mb = new Map({
+    attributionControl: false,
```
Comment (G-Ray) [inline src/adapters/scene.js]: I don't think this is used anywhere ?

File: src/mapbox/extended_attribution_control.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+const { AttributionControl } = require('mapbox-gl--ENV')
+
+/**
+ * Override default control to pass a container
+ * in constructor and register a onReady cb
+ */
+class ExtendedAttributionControl extends AttributionControl {
+  constructor(options, container) {
+    super(options)
+    this.parentCcontainer = container
```
Comment (G-Ray) [inline src/mapbox/extended_attribution_control.js]: typo -> `parentContainer`

File: src/mapbox/extended_scale_control.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+ * in constructor and register a onReady cb
+ */
+class ExtendedScaleControl extends ScaleControl {
+  constructor(options, container) {
+    super(options)
+    this.parentCcontainer = container
+  }
+
+  onAdd(map) {
+    this._map = map;
```
Comment (G-Ray) [inline src/mapbox/extended_scale_control.js]: no semicolons ?

File: src/mapbox/extended_scale_control.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+const { ScaleControl } = require('mapbox-gl--ENV')
+
+/**
+ * Override default control to pass a container
+ * in constructor and register a onReady cb
+ */
+class ExtendedScaleControl extends ScaleControl {
+  constructor(options, container) {
+    super(options)
+    this.parentCcontainer = container
```
Comment (G-Ray) [inline src/mapbox/extended_scale_control.js]: `parentCcccontainer` :rofl:

File: package.json (hunk 3 above)
Comment (Julien-laville) [reply package.json]: sure

File: src/adapters/scene.js (hunk 4 above)
Comment (Julien-laville) [reply src/adapters/scene.js]: yep it is (if not there will be 2 attribute control, the native & the extended)

File: src/mapbox/extended_scale_control.js (hunk 6 above)
Comment (Julien-laville) [reply src/mapbox/extended_scale_control.js]: yep copy pasta 🍝

File: src/mapbox/extended_scale_control.js (hunk 7 above)
Comment (Julien-laville) [reply src/mapbox/extended_scale_control.js]: hahahahahaha 🎮

File: src/mapbox/extended_attribution_control.js (hunk 5 above)
Comment (Julien-laville) [reply src/mapbox/extended_attribution_control.js]: fixed

**Pull request merged**